src/core/auth/qgsauthmanager.cpp

```diff
@@ -790,11 +792,24 @@ bool QgsAuthManager::resetMasterPassword( const QString &newpass, const QString
       *backuppath = dbbackup;
   }
 
+  settingsUsingGeneratedRandomPassword->setValue( false );
+
   QgsDebugMsgLevel( QStringLiteral( "Master password reset: SUCCESS" ), 2 );
   emit authDatabaseChanged();
   return true;
 }
 
+bool QgsAuthManager::resetMasterPasswordUsingStoredPasswordHelper( const QString &newPassword, bool keepBackup, QString *backupPath )
+{
+  if ( !verifyStoredPasswordHelperPassword() )
+  {
+    emit passwordHelperMessageLog( tr( "Master password stored in your %1 is not valid" ).arg( passwordHelperDisplayName() ), authManTag(), Qgis::MessageLevel::Warning );
+    return false;
+  }
+
+  return resetMasterPassword( newPassword, passwordHelperRead(), keepBackup, backupPath );
```

Comment:
if passwordHelperRead() can go wrong perhaps it's better to do the read in a separate instruction and log the error.